framework/src/base/Moose.C

```diff
@@ -462,6 +462,25 @@
 
 #include <unistd.h>
 
+// Define the available execute flags for MOOSE. The flags using a hex value are setup to retain
+// the same numbers that were utilized with the ExecFlagType was a proper enum. This was done for
+// keeping existing applications working using the deprecated flags. In the future, as in the
+// EXEC_SAME_AS_MULTIAPP flag, there is no reason to keep these flags bitwise comparable because
+// the MultiMooseEnum that is used to store these (see SetupInterface) has convenience methods for
+// determining the what flags are active.
+const ExecFlagType EXEC_NONE = 0x00;           // 0
+const ExecFlagType EXEC_INITIAL = 0x01;        // 1
+const ExecFlagType EXEC_LINEAR = 0x02;         // 2
+const ExecFlagType EXEC_NONLINEAR = 0x04;      // 4
+const ExecFlagType EXEC_TIMESTEP_END = 0x08;   // 8
+const ExecFlagType EXEC_TIMESTEP_BEGIN = 0x10; // 16
+const ExecFlagType EXEC_FINAL = 0x20;          // 32
+const ExecFlagType EXEC_FORCED = 0x40;         // 64
+const ExecFlagType EXEC_FAILED = 0x80;         // 128
+const ExecFlagType EXEC_CUSTOM = 0x100;        // 256
+const ExecFlagType EXEC_SUBDOMAIN = 0x200;     // 512
```

Comment:
Right, but I think this is exactly the wrong direction. The bitset idea is actually perfect for execution flags because there's no implied ordering (something MulitMooseEnums gives you). The real beauty in using an actual flags set is you have a single scalar type that represents all possible flags so every single vector and set that you use throughout this PR just goes away.